lib/pwwka/transmitter.rb

```diff
@@ -25,17 +30,38 @@ def initialize
     # routing_key:: String routing key for the message
     # delayed:: Boolean send this message later
     # delay_by:: Integer milliseconds to delay the message
+    # on_error:: What is the behavior of
+    # - :ignore (aka as send_message_safely)
+    # - :raise
+    # - :resque -- use Resque to try to send the message later
     #
     # Returns true
     #
     # Raises any exception generated by the innerworkings of this library.
-    def self.send_message!(payload, routing_key, delayed: false, delay_by: nil)
+    def self.send_message!(payload, routing_key, on_error: :raise,
+        delayed: false, delay_by: nil)
       if delayed
         new.send_delayed_message!(*[payload, routing_key, delay_by].compact)
       else
         new.send_message!(payload, routing_key)
       end
       info "AFTER Transmitting Message on #{routing_key} -> #{payload}"
+    rescue => e
+      error "ERROR Transmitting Message on #{routing_key} -> #{payload}: #{e}"
+      case on_error
+        when :raise
+          raise e
+        when :resque
+          send_message_async(payload, routing_key, delay_by_ms: delayed ? delay_by || DEFAULT_DELAY_BY_MS : 0)
+        else # ignore
+      end
+      false
+    end
+
+    # Use Resque to enqueue the message.
+    # - :delay_by_ms:: Integer milliseconds to delay the message. Default is 0.
+    def self.send_message_async(payload, routing_key, delay_by_ms: 0)
+      Resque.enqueue_in(delay_by_ms/1000, SendMessageAsyncJob, payload, routing_key)
```

Comment:
I'm not sure what you're asking. Is it what happens if Resque is fails? It will raise the Resque error.

We thought through the RabbitMQ outage, using Resque as a backup. That would be how we used it, and wouldn't call this method directly. We debated just not having it, and I'm not totally convinced we need it.